### ISSUE TITLE
Bug in KnownTypesDataContractResolver

### ISSUE THREAD

**Issue by maulwuff** [dotnetjunkie/solidservices#16]
I've been playing around with the wcf client and found, that the knowTypes detection is too aggressive:

```csharp
 this.knownTypes = (
                from type in types.Distinct()
                group type by type.Name into g
                select g.First())
                .ToDictionary(GetName);
````
This line is faulty: `group type by type.Name into g` 

`type.Name` causes types like the following to be treated as the same:
`GetUserRightsQuery : IQuery<Paged<UserRight>>`
`GetUsersQuery : IQuery<Paged<User>>`
because type.Name returns `Paged1` (or similar) for both of them.

What was the intention of the grouping? 
When I remove `.Name` I get all necessary types to deserialize the wcf result.

**Comment by dotnetjunkie** [comment]
I think you are right. I'm not sure what problem I thought of solving with that. As far as I can see, the correct code is:

``` c#
this.knownTypes = types.Distinct().ToDictionary(GetName);
```

**Comment by dotnetjunkie** [comment]
Fixed in cf840bf2a346be60088e83c95e8a0cee28c9c054
